### ISSUE TITLE
Gitea: use repo topic as a cache

### ISSUE THREAD

**Issue by stevefan1999-personal** [renovatebot/renovate#18921]
### What would you like Renovate to be able to do?

I want to use the autodiscovery feature but I don't want to pull the repo every single time, causing a huge spike in our potato server. I don't want to use a cache either because there can be multiple instances of Renovate running at the same time (for different repo too) and using a consolidated cache would cause some synchronization issue. We are using a K8S cronjob for this so we can't afford to update the repo every time we make a new one. Obviously, we wanted a more dynamic choice.

This would greatly reduce the need for manually configuring the repos, while still keeping a clear intention to the rest of my teammates.

### If you have any ideas on how this should be implemented, please tell us here.

We can use the [Gitea topic support](https://github.com/go-gitea/gitea/pull/3711) to filter projects that are manually tagged to want to onboard, like if the repo is tagged with `renovate-managed`.

If the filtered repo with topic does not have the config.json/config.js, it is considered to be in onboard mode. 

Say if I have a `stevefan1999/foo` repo tagged with `renovate-managed`, and Renovate only clones that, tried to scan but not see `renovate.json`/`renovate.js`, we send a `Configure Renovate`/onboarding PR for that.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by viceice** [comment]
so basically you like to use a tag to disable repos from being processed by renovate.

so this can maybe implemented like the experimental autodiscover server filter I've done recently.

**Comment by stevefan1999-personal** [comment]
@viceice Quite close -- except I want to manually cherrypick the repos I want to auto-discover on

**Comment by viceice** [comment]
🤔 @stevefan1999-personal Please try to describe in more detail how you think this should work.

Renovate currently does this:
1. autodiscover repos, if enabled
2. apply autodiscover filter
3. append statically configured repos (filter duplicates)
4. process each repo

**Comment by stevefan1999-personal** [comment]
@viceice Yes I know, except I don't want to auto discover too much repo -- right now we have over 100 repos and I don't expect to clone them one by one because it will thwart our Gitea instance. 

I want to use the topic instead to act as a filter without having to set the `config.js`/`config.json` every time to update the autodiscovery filter, since this not only would mean I have to update the Helm chart a lot (we use https://github.com/drone/charts btw), but we can't add it to the autodiscovery filter on demand sometimes due to permission issue that not all of us have RBAC permission to update the Helm chart and thus the filter while the audit team members are away. 

This RBAC permission problem is specifically for us, so just think of it as a little shortcut for probing the candidates of autodiscovery targets by leveraging the topic feature to eagerly filter the search space early on.

**Comment by viceice** [comment]
ok, so what you like is to add a new topic filter here:
https://github.com/renovatebot/renovate/blob/3c56aeca17fb00c72f3546c8077c7b23b01f8118/lib/modules/platform/gitea/index.ts#L349-L354

So a repo needs a specific topic to be returned if it's defined.

**Comment by viceice** [comment]
checkout this pr for implementation details
- #18636

**Comment by stevefan1999-personal** [comment]
@viceice Thanks! Actually I did look up there before and know what to do but I'm gotta rest for now
